src/votekit/ballot_generator.py

```diff
@@ -1645,21 +1693,48 @@ def __init__(self, cohesion_parameters: dict, **data):
                               passed {len(self.slate_to_candidates.keys())}"
             )
 
-        self.ballot_type_pdf = {
-            b: self._compute_ballot_type_dist(b, self.blocs[(i + 1) % 2])
-            for i, b in enumerate(self.blocs)
-        }
+        if len(self.candidates) < 12 and len(self.blocs) == 2:
+            # precompute pdfs for sampling
+            self.ballot_type_pdf = {
+                b: self._compute_ballot_type_dist(b, self.blocs[(i + 1) % 2])
+                for i, b in enumerate(self.blocs)
+            }
+
+        elif len(self.blocs) == 1:
+            # precompute pdf for sampling
+            # uniform on ballot types
+            bloc = self.blocs[0]
+            bloc_to_sample = [
+                bloc
+                for _ in range(
+                    len(self.pref_intervals_by_bloc[bloc][bloc].non_zero_cands)
+                )
+            ]
+            pdf = {tuple(bloc_to_sample): 1}
+            self.ballot_type_pdf = {bloc: pdf}
+
+        else:
+            warnings.warn(
```

Comment:
Should this be an error?

Reply:
I don't think so; it's totally fine to have two blocs and more than 11 candidates, you just can't use the deterministic generating method. You can use the MCMC one.